drivers/soundwire/stream.c

```diff
@@ -267,6 +267,13 @@ static int sdw_program_port_params(struct sdw_master_runtime *m_rt)
 	/* Program transport & port parameters for Slave(s) */
 	list_for_each_entry(s_rt, &m_rt->slave_rt_list, m_rt_node) {
 		list_for_each_entry(p_rt, &s_rt->port_list, port_node) {
+			/*
+			 * skip unprepared slaves since transport & port
+			 * parameters are not computed for them
+			 */
+			if (!p_rt->num)
+				continue;
+
```

Comment:
@RanderWang I don't think this is the right fix, it looks instead as if the data structures are incorrect.

We should have a master_runtime per stream, and then the list of slaves that are used, and last the ports. So when we program the port parameters, we should be able to filter out the entire master runtime.

Also in some cases we will use Port0 for transmission of audio data in BRA mode, so we should not test for port_num != 0.

What you've done is very good, finding the root cause was a very significant progress, but this fix can be improved further.

Reply:
@pierre   please check the following description https://github.com/thesofproject/linux/blob/integration/soundwire-intel/drivers/soundwire/bus.h#L81
````
* @num: Port number. For audio streams, valid port number ranges from [1,14]
````
I need your help to make it clear.  Thanks!

Reply:
I want to get a elegant method to address this issue. First, I added a flag to indicate whether the port is initialized or not. After discussing it with bard,  I abandoned it and also many other changes. I will update my PR. Thanks for review.